api/requests/urls.py

```diff
@@ -0,0 +1,8 @@
+from django.conf.urls import url
+
+from . import views
+
+urlpatterns = [
+    url(r'^(?P<request_id>\w+)/$', views.NodeRequestDetail.as_view(), name=views.NodeRequestDetail.view_name),
+    url(r'^(?P<request_id>\w+)/actions/$', views.CreateNodeRequestAction.as_view(), name=views.CreateNodeRequestAction.view_name),
```

Comment:
Related: If the above was the case, I don't think the `/requests/` endpoints would be needed (all operations would be handled within `/actions/` or `nodes/<id>/requests/`).

Reply:
Also, it seems like you meant to make requests a general model (`AbstractRequest`), but these endpoints seem to only handle a specific type of request (`NodeRequest`). 

Do requests need to be this abstracted? What are the future use cases? If so, I think these endpoints should be generalized a bit more.

Reply:
>It seems that most `BaseAction` operations should be handled within the `/actions/` endpoint. [...]

When this was written, the `Generalize Actions` ticket had received cross-team discussion, and it was agreed that related views (e.g. `/requests/<id>/actions`) were preferable to sub-routes (`actions/requests/`). Some time later, this decision was reversed, and this PR hadn't been updated from that. It would make sense to do that now.

>  I don't think the `/requests/` endpoints would be needed

You're probably right about that... at least until other types of `AbstractRequest` are added.

>Do requests need to be this abstracted? 

According to all the discussion on this (from months ago), yes.

>What are the future use cases?

Other things users might "request"? -- "Unlock this file", "Let me update this file without having full-write", "Share this private (non-node) thing with me" ¯\\\_(ツ)_/¯ 
All of these are guesses, and I'm sure there are things no one has thought of yet. To fully answer this question, you'd need a 🔮 

>If so, I think these endpoints should be generalized a bit more.

The `/actions/` sub-route and related view through `/nodes/` should accomplish that.